streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -321,27 +348,31 @@ class ZStream[-R, +E, +A](val channel: ZChannel[R, Any, Any, Any, E, Chunk[A], A
 
   /**
    * Allows a faster producer to progress independently of a slower consumer by buffering
-   * up to `capacity` elements in a dropping queue.
+   * up to `capacity` chunks in a dropping queue.
    *
    * @note Prefer capacities that are powers of 2 for better performance.
    */
-  final def bufferDropping(capacity: Int): ZStream[R, E, A] =
-    ???
+  final def bufferChunksDropping(capacity: Int): ZStream[R, E, A] = {
+    val queue = Queue.dropping[(Take[E, A], Promise[Nothing, Unit])](capacity).toManaged(_.shutdown)
+    new ZStream(bufferSignal[R, E, A](queue, _.upstream))
+  }
 
   /**
    * Allows a faster producer to progress independently of a slower consumer by buffering
-   * up to `capacity` elements in a sliding queue.
+   * up to `capacity` chunks in a sliding queue.
    *
    * @note Prefer capacities that are powers of 2 for better performance.
    */
-  final def bufferSliding(capacity: Int): ZStream[R, E, A] =
-    ???
+  final def bufferChunksSliding(capacity: Int): ZStream[R, E, A] = {
+    val queue = Queue.sliding[(Take[E, A], Promise[Nothing, Unit])](capacity).toManaged(_.shutdown)
+    new ZStream(bufferSignal[R, E, A](queue, _.upstream))
+  }
 
   /**
    * Allows a faster producer to progress independently of a slower consumer by buffering
-   * elements into an unbounded queue.
+   * chunks into an unbounded queue.
    */
-  final def bufferUnbounded: ZStream[R, E, A] = {
+  final def bufferChunksUnbounded: ZStream[R, E, A] = {
```

Comment:
Hmm. We're buffering with an unbounded size - so there's no real difference between chunked/unchunked no? This could be the only operator we have.

Reply:
Makes sense, keeping just one of these then.

Reply:
Done